pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -386,7 +386,7 @@ private void readMorePolicies(SystemTopicClient.Reader<PulsarEvent> reader) {
                 notifyListener(msg);
                 readMorePolicies(reader);
             } else {
-                if (ex instanceof PulsarClientException.AlreadyClosedException) {
+                if (ex.getCause() instanceof PulsarClientException.AlreadyClosedException) {
```

Comment:
According to context, I think there don't need to ``getCause``
```java
    @Override
    public CompletableFuture<Message<T>> receiveAsync() {
        if (listener != null) {
            return FutureUtil.failedFuture(new PulsarClientException.InvalidConfigurationException(
                    "Cannot use receive() when a listener has been set"));
        }
        try {
            verifyConsumerState();
        } catch (PulsarClientException e) {
            return FutureUtil.failedFuture(e);
        }
        return internalReceiveAsync();
    }
```
because the exception created by ``CompletableFuture#completeExceptionally``, in this case, we don't need to ``getCause``.
for example: 

```java
class Scratch {
    public static void main(String[] args) {
        CompletableFuture<Void> future = new CompletableFuture<>();
        future.completeExceptionally(new RuntimeException("123"));
        future.whenComplete((value, ex) -> {
            if (ex instanceof RuntimeException) {
                System.out.println("runtime exception");
            } else {
                System.out.println("completion exception");
            }
        });
    }
}
```

> runtime exception
Process finished with exit code 0

Thanks and please let me know what you think.

Reply:
No, I don't think so.  The test could verify this.

Reply:
The stack trace if here :
```
2022-03-27T10:42:16,795+0800 [broker-client-shared-internal-executor-58-1] WARN  org.apache.pulsar.broker.service.SystemTopicBasedTopicPoliciesService - Read more topic polices exception, read again.
java.util.concurrent.CompletionException: org.apache.pulsar.client.api.PulsarClientException$AlreadyClosedException: Consumer already closed
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292) ~[?:1.8.0_291]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308) ~[?:1.8.0_291]
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:607) ~[?:1.8.0_291]
	at java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:628) ~[?:1.8.0_291]
	at java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:1996) ~[?:1.8.0_291]
	at org.apache.pulsar.client.impl.MultiTopicsReaderImpl.readNextAsync(MultiTopicsReaderImpl.java:140) ~[classes/:?]

```

Reply:
CompletionException is subclass of RuntimeException

Reply:
Sorry for the imprecise demo, I paste new demo here:

```java
class Scratch {
    public static void main(String[] args) {
        CompletableFuture<Void> future = new CompletableFuture<>();
        future.completeExceptionally(new IllegalArgumentException("123"));
        future.whenComplete((value, ex) -> {
            if (ex instanceof IllegalArgumentException) {
                System.out.println("illegal exception");
            } else {
                System.out.println("completion exception");
            }
        });
    }
}
```

Reply:
After discussing, it's safe to use `Throwable cause = FutureUtil.unwrapCompletionException(ex);`

Reply:
**Supplementary context (after discuss with author)**

Due to the inconsistent behaviour of ``readNextAsync`` regarding ``ReaderImpl`` and ``MultipleReaderImpl`` , we can use ``Throwable cause = FutureUtil.unwrapCompletionException(ex);`` to compatible.